tests/test_ai_sidecar_external.py

```
@@ -0,0 +1,204 @@
+    assert "unknown action" in err["message"]
+
+
+def test_config_set_round_trip_via_hub() -> None:
+    """Two peers: A sends config.set, B receives it; B's ack reaches A."""
+
+    async def _run() -> tuple[dict, dict]:
+        async with _running_sidecar() as port:
+            async with (
+                ws_connect(f"ws://127.0.0.1:{port}/") as a,
+                ws_connect(f"ws://127.0.0.1:{port}/") as b,
+            ):
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)